### PR TITLE
PKF 430 Phase 1: system deep sleep mode

### PR DESCRIPTION
**Pull request by minchingho-pax**
This change can not be verified by itself. A separated pull request will be created for pax-common-code.

### CODE REVIEW

File: hw/drivers/sensors/lis2dw12/src/lis2dw12.c [hunk 1]
```diff
@@ -3699,3 +3700,21 @@ lis2dw12_create_spi_sensor_dev(struct bus_spi_node *node, const char *name,
...
+int
+lis2dw12_power_control(struct sensor *sensor, void *args)
+{
+    struct sensor_itf *itf;
+    int power_state = *(int *)args;
+
+
+    itf = SENSOR_GET_ITF(sensor);
+
+    if(power_state == 0)
```
Comment (minchingho-pax) [inline hw/drivers/sensors/lis2dw12/src/lis2dw12.c]: This state is not fully defined yet. 0 means off. I will create a enum in the near future to cover this.

Comment (pradyumnapax) [reply hw/drivers/sensors/lis2dw12/src/lis2dw12.c]: Thanks

File: hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c [hunk 2]
```diff
@@ -652,7 +652,7 @@ hal_timer_config(int timer_num, uint32_t freq_hz)
     hwtimer = bsptimer->tmr_reg;
 
     /* Stop the timer first */
-    hwtimer->TASKS_STOP = 1;
+    hwtimer->TASKS_CLEAR = 1;
```
Comment (pradyumnapax) [inline hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c]: why clear timer not stop timer?. Can you please add comments or modify above comments.

File: hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c [hunk 3]
```diff
@@ -703,7 +703,7 @@ hal_timer_deinit(int timer_num)
     } else {
         hwtimer = (NRF_TIMER_Type *)bsptimer->tmr_reg;
         hwtimer->INTENCLR = NRF_TIMER_INT_MASK(NRF_TIMER_CC_INT);
-        hwtimer->TASKS_STOP = 1;
+        hwtimer->TASKS_SHUTDOWN = 1;
```
Comment (pradyumnapax) [inline hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c]: Nordic datasheet shows TASKS_SHUTDOWN as deprecated, any idea about this?. What is the difference between both, don't see much info in datasheet. Can you please add comment if possible?.

Comment (pradyumnapax) [reply hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c]: Ignore my comments I guess. As cherry-picked from mynewt.

**Pull request merged**